### ISSUE TITLE
Ethers.js ABI Module / EthereumJS ABI Library - Deprecation Analysis

### ISSUE THREAD

**Issue by holgerd77** [ethereum/js-team-organization#3]
Both Ethers.js and EthereumJS have a module respective library for ABI encoding and decoding, see:

- [Ethers.js ABI Coder](https://github.com/ethers-io/ethers.js/blob/master/src.ts/utils/abi-coder.ts)
- [ethereumjs-abi Library](https://github.com/ethereumjs/ethereumjs-abi)

This issue is for a first exemplary analysis if one of these libraries can be deprecated and/or slowly replaced by the respective other. Therefore there should be some comparison on the following aspects (feel free to edit or add):

- Functionality
- Code & Documentation Quality
- Test Coverage and Security
- User base
- Targeted Environments / Other limitations

**Comment by ricmoo** [comment]
One thing I'd like to try again (it's been a few years) is running the EthereumJS coder against the ethers.js Test Suite. Last I tried (again, ages ago) about 40% of the test cases failed. I was able to track one of the main issues to be the conversion of `int` to something like `nt256` because of a copy-paste issue when handling the default size types from `uint`.

But it would be good to run each other's test suite against the other's anyways. :)
